### ISSUE TITLE
Allow Dataverse Admins to Save Preferred Sort Order

### ISSUE THREAD

**Issue by posixeleni** [IQSS/dataverse#3066]
By default all datasets in a Dataverses are sorted by Newest with the option to change to other options. However, these options cannot be saved for a particular Dataverse.

**Comment by pdurbin** [comment]
Wouldn't that be a little jarring for people browsing the site? What if instead users could set a preference in their account?

In practice there are only two ways to sort: newest/oldest and alphabetically:

![demo_dataverse_-_2016-04-07_10 44 27](https://cloud.githubusercontent.com/assets/21006/14355118/dc8f8194-fcad-11e5-987f-b0f9baf4457c.png)

I feel like this is at least somewhat related to how people are having a little trouble figuring out what the "tree" of dataverses within an installation looks like: https://github.com/IQSS/dataverse/issues/752#issuecomment-203525523 

This confusion is a real issue. I'm just not sure about this suggestion.

**Comment by posixeleni** [comment]
@pdurbin I am not suggesting anywhere in particular for the where this should be set in the UI since this is really up to the UI/UX team to decide where the ideal place to set this would be. I just wanted to create this ticket as a placeholder.

**Comment by tlchristian** [comment]
This is something that some of our users would like, also.  On their Dataverses, they would like to sort by alpha.  They even named their datasets in such a way that a default sort by alpha would make the most sense for their users.  It would be nice if we can change the default sort setting.

**Comment by pdurbin** [comment]
The customer is always right, I guess. Thanks for the feedback @tlchristian :)

**Comment by pdurbin** [comment]
@tlchristian we discussed this issue briefly yesterday during a sprint planning meeting and we're planning on getting in touch with you to get a better understanding of what problem you're trying to solve.

**Comment by jggautier** [comment]
Thanks @parsr for sharing, [in another issue](https://github.com/IQSS/dataverse/issues/2297#issuecomment-341147484), about a change to sorting by date:

> Then you also have the sorting issue - we want the most recent ("newest") dataset on top but adding previous years' datasets messes with the the sort order (which should be by "newest" by true publication date not "newest" in terms of being added to dataverse).
> 
> Very annoying problem - when we're trying to release something "new" while at the same time add older datasets to a dataverse. Everything appears "new" but only one dataset is published in current year.

**Comment by jggautier** [comment]
Since at least Dataverse 4, when sorting by "Newest", datasets are sorted by the date of its newest published major version. (Dataverses are sorted by when they are published.)

@parsr is it fair to say you'd prefer to sort by the date of the first saved version of the dataset?

**Comment by pdurbin** [comment]
> datasets are sorted by the date of its newest published major version

Yep, this was implemented in #287 and the design doc from back then is https://docs.google.com/a/harvard.edu/document/d/1DWsEqT8KfheKZmMB3n_VhJpl9nIxiUjai_AIQPAjiyA/edit?usp=sharing

**Comment by parsr** [comment]
@jggautier Not first saved version, but on asserted publication date.

For new, never published datasets, published date is / can be when you hit the Publish button.  But for migrated datasets (already published), it makes sense to assert the real published date from past event and be able to rely on that for ordering.

Here's a sample listing of use case scenario I'm talking about: [http://www.cmcrp.org/media-industry-data/archived-data/](http://www.cmcrp.org/media-industry-data/archived-data/)  

Job is to migrate them into our dataverse, but older datasets published in previous years shouldn't all show up cited as "2017" for PY in the citation just because they were added to dataverse this year.  There's a new one coming in a couple of weeks.  That upcoming one should be on top of the dataverse listing, but looks like if we add all those archived datasets, the new one could get lost amongst all the them since they'll all be "2017" and/or sorted "newest" by entry  date (e.g. when we hit "publish").

So would I have to make sure I add those archived datasets oldest to newest, and get them in before this years release, to get this year's release to be on top of the dataverse listing?

will look at the API option

**Comment by pdurbin** [comment]
> will look at the API option

The multiple issues are a bit hard to follow but I assume this means https://github.com/IQSS/dataverse/issues/2297#issuecomment-341250897 😄

**Comment by jggautier** [comment]
Thanks for the example dataset @parsr, although I keep getting an error that the site can't be reached.

What we're talking about is changing what date the "Newest"/"Oldest" sort uses (or, another option, adding a second way to sort by a different type of date), so it also might be outside of the scope of this issue, which is about setting a different default sort, but I didn't want to create a third related issue, because it does sound like @parsr would also like the default sort to change (unless the date used for the "Newest"/"Oldest" sort is changed to the date when the dataset was first published in any system, in which case "Newest"/"Oldest" can remain the default).

Dataverse has a "Distribution Date" field that I think is appropriate for "the real published date."

> Distribution Date: Date that the data was made available for distribution/presentation.

One option might be to put in the Distribution Date field the date when your datasets were published in another system, and then either:

1 - Change how the "Newest" and "Oldest" sort works so that they're based on the date in the Distribution Date field, or, if there's no Distribution Date (such as for datasets first published in Dataverse), sorted by the system-generated "Publication Date" (defined as the date the dataset was first published in Dataverse)

2 - Allow users to sort by two different dates:
- newest major version date (behaving the way it does now)
- "Distribution Date" or, when there's no Distribution Date", "Publication Date"

make the labels on the drop down buttons explicit, e.g. "Newest (Distribution Date)", "Newest (Version)", and make one of those the default sort.

**Comment by pdurbin** [comment]
Related: #5131

**Comment by pdurbin** [comment]
We should probably close this issue in favor of this one:

- #7855

**Comment by pdurbin** [comment]
Closing in favor of this:

- #7855
